tax-processing-functions/quality_and_split/main.py

```diff
@@ -0,0 +1,403 @@
+bq_client = bigquery.Client()
+publisher = pubsub_v1.PublisherClient()
+ts_format = "%Y-%m-%d_%H:%M:%S"
+
+
+# Entry point for Cloud Function
+def process_receipt(event, context):
+    # event = cloud function event
+    # global input_bucket_name
+    # global file_name
+    # global content_type
+    input_bucket_name = event["bucket"]
+    file_name = event["name"]
+    content_type = event["contentType"]  # TODO: rewrite
+    print(f"Found content type:{content_type}")
+
+    if content_type in accepted_file_types:
+        # Gets file blob
+        file = get_gcs_file(file_name, input_bucket_name)
+
+        # Extracts entities from Document AI
+        extracted_doc = extract_entities(file, content_type, processor_quality)
+
+        # Splitter--function, parse quality metrics
+        avg_q = parse_quality_metrics(extracted_doc)  # IDQ JSON Output (Quality)
+
+        # Separate documents by quality threshold
+        if avg_q > 0.05:
+            # Extract document entities using classification processor
+            extracted_classifier_doc = extract_entities(
+                file, content_type, processor_classification
+            )
+
+            if DEBUG:
+                print(extracted_classifier_doc)  # Splitter/classifier JSON output
+
+            # Parse documents
+            parse_doc_type(
+                extracted_classifier_doc,
+                file,
+                file_name,
+                content_type,
+                workflow_topic_name,
+                input_bucket_name,
+            )
+
+        else:
+            print("Fail, image qualtity is bad")
+            # Extract entities and push to pubsub for review
+            extracted_classifier_doc = extract_entities(
+                file, content_type, processor_classification
+            )
+            extracted_noid_doc_processed = format_entities(extracted_classifier_doc)
+            json_string = json.dumps(extracted_noid_doc_processed, sort_keys=False)
+            topic_path = publisher.topic_path(project_id, uil_review_topic_name)
+            publish_to_pubsub(topic_path, json_string)
+
+    else:
+        print(f"Cannot parse the file type. Acceoted types are {accepted_file_types}")
+        # Move file to rejected GCS bucket
+        move_blob(input_bucket_name, bucket_rejected, blob_name=file_name)
+        return
+
+
+# Splitter function - return the average quality rating for document entities
+# DISCLAIMER: this returns an average quality of entities in the document
+def parse_quality_metrics(raw_content):
+    # raw_content = entities extracted from DocAI processor
+    if raw_content:
+        num_pages = len(raw_content.entities)
+        total = 0
+        for entity in raw_content.entities:
+            if str(entity.type_) == "quality_score":
+                total += entity.confidence
+        avg_q = total / num_pages
+        print(f"Average quality in document entities: {avg_q}")
+        return avg_q
```

Comment:
This can also be changed to a "guard statement" to reduce indentation.
I'd also recommend changing the variable name to be `document` or just pass `entities` as the parameter.